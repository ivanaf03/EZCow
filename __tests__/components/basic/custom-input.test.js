import React from "react";

import {render, fireEvent} from "@testing-library/react-native";

import CustomInput from "../../../components/basic/custom-input";

describe("CustomInput", () => {

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("should render correctly", () => {
        const tree = render(<CustomInput text="Test" placeholder="Test" value="Test" />);
        expect(tree).toMatchSnapshot();
    });

    it("should change value when typing", () => {
        const onChangeText = jest.fn();
        const tree = render(<CustomInput text="Test" placeholder="Test" value="Test" onChangeText={onChangeText} />);
        fireEvent.changeText(tree.getByPlaceholderText("Test"), 'Test2');
        expect(onChangeText).toHaveBeenCalledTimes(1);
    });

});