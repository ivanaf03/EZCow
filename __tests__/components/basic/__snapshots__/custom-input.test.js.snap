// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput should render correctly 1`] = `
<View
  style={
    [
      {
        "marginTop": -4,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 14,
          "lineHeight": 20,
        },
        {
          "fontFamily": "Nunito-Medium",
        },
        [
          {
            "marginTop": 4,
          },
        ],
      ]
    }
  >
    Test
    :
  </Text>
  <TextInput
    placeholder="Test"
    style={
      [
        {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
        {
          "backgroundColor": "#ffffff",
        },
        {
          "borderBottomWidth": 2,
          "borderLeftWidth": 2,
          "borderRightWidth": 2,
          "borderTopWidth": 2,
        },
        {
          "borderBottomColor": "#2254ba",
          "borderLeftColor": "#2254ba",
          "borderRightColor": "#2254ba",
          "borderTopColor": "#2254ba",
        },
        {
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
        },
        {
          "fontFamily": "Nunito-Medium",
        },
        [
          {
            "marginTop": 4,
          },
        ],
      ]
    }
    value="Test"
  />
</View>
`;
